frappe/commands/site.py

```diff
@@ -31,10 +31,10 @@
 )
 @click.option("--db-root-password", "--mariadb-root-password", help="Root password for MariaDB or PostgreSQL")
 @click.option(
-	"--no-mariadb-socket",
```

Comment:
Better to leave this backward compatible, it seems to be used in enough places: https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/frappe/.*+--no-mariadb-socket&patternType=standard&sm=0&groupBy=repo

Reply:
I'm putting back the flag so that the command line wouldn't complain.

However I have a hard time seeing how the only two consumers of `host = '%'`:
```python
	def create_user(self, user, password, host=None):
		host = host or self.get_current_host()
		password_predicate = f" IDENTIFIED BY '{password}'" if password else ""
		self.db.sql(f"CREATE USER IF NOT EXISTS '{user}'@'{host}'{password_predicate}")
```

```python 
	def grant_all_privileges(self, target, user, host=None):
		host = host or self.get_current_host()
		permissions = (
			(
				"SELECT, INSERT, UPDATE, DELETE, CREATE, DROP, INDEX, ALTER, "
				"CREATE TEMPORARY TABLES, CREATE VIEW, EVENT, TRIGGER, SHOW VIEW, "
				"CREATE ROUTINE, ALTER ROUTINE, EXECUTE, LOCK TABLES"
			)
			if frappe.conf.rds_db
			else "ALL PRIVILEGES"
		)
		self.db.sql(f"GRANT {permissions} ON `{target}`.* TO '{user}'@'{host}'")
```

would differ when:

```python
	def get_current_host(self):
		return self.db.sql("select user()")[0][0].split("@")[1]
```

is replaced by `%`

Reply:
Ok, I finally got to the bottom of this in ca6176da92

Based on: https://dev.mysql.com/doc/refman/8.0/en/account-names.html

Note:

> The @'host_name' part is optional. An account name consisting only of a user name is equivalent to 'user_name'@'%'. For example, 'me' is equivalent to 'me'@'%'.

And:

> Use of the % and _ wildcard characters is permitted in host name or IP address values, but is deprecated as of MySQL 8.0.35, and thus subject to removal in a future version of MySQL.

Reply:
@akhilnarang would you agree with these changes? Should we mark this as resolved, as well?

Reply:
Will review and test soon